File: pa3/DictionaryTest.java
// Austin Yen
// 4/28/15
// DictionaryTest.java
// Dictionary ADT test file, individually testing the 
// methods and also during different states of a 
// dictionary linked list
public class DictionaryTest {
	public static void main(String[] args){
		
		// create new Dictionary
		Dictionary dict = new Dictionary();
		
		
		// Check if isEmpty() returns true
		// System.out.println(dict.isEmpty()); 
		// It returned true
		
		// Check if size() returns 0
		// System.out.println(dict.size()); 
		// It returned 0
		
		// Try inserting some values
		dict.insert("1","a");
		dict.insert("2","b");
		dict.insert("3","c");
		dict.insert("4","d");
		dict.insert("5","e");
		dict.insert("6","f");
		// check if everything prints as specified in toString()
		// 1 a
		// 2 b
		// 3 c
		// ...
		// System.out.print(dict.toString());
		// everything printed correctly
		
		// Check if size() returns the proper size
		// System.out.println("size = "+dict.size()); 
		// It returned 6, which is correct
		

		
		
		// check if delete() removes the item specified by the key
		dict.delete("1"); 
		dict.delete("3");
		dict.delete("6");
		// check if the delete method works by printing
		System.out.print(dict.toString()); 
		// it printed everything except the specified items
		
		// Check again if size() returns the proper size
		System.out.println("size = "+dict.size()); 
		// It returned 3, which is correct
		
		// check if lookup finds the key and returns value
		// System.out.println("value of 1 is "+dict.lookup("1")); 
		// lookup() returned "a"
		
		
		// check if isEmpty() returns false
		// System.out.println(dict.isEmpty()); 
		// It returned false
		
		// try the makeEmpty() method
		// makeEmpty() works if isEmpty() returns true
		dict.makeEmpty(); 
		System.out.println(dict.isEmpty()); 
		// isEmpty returned true
		
		
		// try adding something to check again
		// should print out the key and value of only what was added
		dict.insert("1","a"); 
		System.out.print(dict.toString()); 
		// toString printed out "1 a"
	}
}

File: lab5/DictionaryTest.c
//-----------------------------------------------------------------------------
// Austin Yen
// 5/9/15
// Testing functions for the Dictionary ADT
// DictionaryTest.c
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary D = newDictionary();

   // Check if isEmpty() returns true
   // printf("empty? %s\n", (isEmpty(D) ? "true" : "false"));
   // It returned true

   // Check if size() returns 0
   // printf("size = %d\n", size(D));
   // It returned 0

   // Try inserting some values
   insert(D,"1", "a");
   insert(D, "2", "b");
   insert(D, "3", "c");
   insert(D, "4", "d");
   insert(D, "5", "e");
   insert(D, "6", "f");

   // try the print function
   // printDictionary(stdout, D);

   // see if size() returns the proper number
   //printf("size = %d\n", size(D)); // printed "size = 6"

   // check delete function using size and print functions
   // delete(D, "2");
   // delete(D, "5");
   delete(D, "1");
   printf("size = %d\n", size(D)); // printed "size = 3"
   printDictionary(stdout, D); // printed correctly

   // check if lookup finds a value by key
   printf("%s\n", lookup(D, "3")); // printed "c"

   // Check if isEmpty() returns false
   printf("empty? %s\n", (isEmpty(D) ? "true" : "false")); // returned false

   // check makeEmpty()
   // use isEmpty() to check
   printf("make empty\n");
   makeEmpty(D);
   printf("empty? %s\n", (isEmpty(D) ? "true" : "false")); // returned true

   // try adding something to check again
   // should print out the key and value of only what was added
   // insert(D,"1", "a"); 
   // printDictionary(stdout, D); // printed "1 a"

   freeDictionary(&D);
   return(EXIT_SUCCESS);
}
